examples/combined_ODNP.py

```diff
@@ -57,54 +51,101 @@
-IR_ph2_cyc = psp.r_[0, 2]
-date = datetime.now().strftime("%y%m%d")
+# }}}
+#{{{phase cycling
+Ep_ph1_cyc = r_[0, 1, 2, 3]
+total_points = len(Ep_ph1_cyc)*nPoints
+assert total_points < 2**14, "For Ep: You are trying to acquire %d points (too many points) -- either change SW or acq time so nPoints x nPhaseSteps is less than 16384"%total_pts
+IR_ph1_cyc = r_[0, 2]
+IR_ph2_cyc = r_[0, 2]
+total_pts = len(IR_ph2_cyc)*len(IR_ph1_cyc)*nPoints
+assert total_pts < 2**14, "For IR: You are trying to acquire %d points (too many points) -- either change SW or acq time so nPoints x nPhaseSteps is less than 16384"%total_pts
+#}}}
 # {{{ check for file
-myfilename = date + "_" + output_name + ".h5"
-if os.path.exists(myfilename):
+if os.path.exists(filename_out):
     raise ValueError(
-        "the file %s already exists, so I'm not going to let you proceed!" % myfilename
+        "the file %s already exists, so I'm not going to let you proceed!" % filename_out
     )
+input("B12 needs to be unplugged and turned off for the thermal! Don't have the power server running just yet")
+# }}}
+# {{{Collect Thermals - serves as a control to compare the thermal of Ep to ensure no microwaves were leaking
+control_thermal = run_spin_echo(
+    nScans=parser_dict['thermal_nScans'],
+    indirect_idx=0,
+    indirect_len=len(powers) + 1,
+    ph1_cyc=Ep_ph1_cyc,
+    adcOffset=parser_dict['adc_offset'],
+    carrierFreq_MHz=parser_dict['carrierFreq_MHz'],
+    nPoints=nPoints,
+    nEchoes=parser_dict['nEchoes'],
+    p90_us=parser_dict['p90_us'],
+    repetition=parser_dict['repetition_us'],
+    tau_us=parser_dict['tau_us'],
+    SW_kHz=parser_dict['SW_kHz'],
+    output_name=filename,
+    indirect_fields=("start_times", "stop_times"),
+    ret_data=None,
+)  # assume that the power axis is 1 longer than the
+#                         "powers" array, so that we can also store the
+#                         thermally polarized signal in this array (note
+#                         that powers and other parameters are defined
+#                         globally w/in the script, as this function is not
+#                         designed to be moved outside the module
+control_thermal = control_thermal['nScans',-1:]
+control_thermal.set_prop("postproc_type", "spincore_ODNP_v3")
+control_thermal.set_prop("acq_params", parser_dict.asdict())
+control_thermal.chunk("t", ["ph1", "t2"], [len(Ep_ph1_cyc), -1])
+control_thermal.setaxis("ph1", Ep_ph1_cyc / 4)
+control_thermal.setaxis('nScans',r_[0:parser_dict['nScans']])
+control_thermal.reorder(['ph1','nScans','t2'])
+control_thermal.ft('t2',shift=True)
+control_thermal.ft(['ph1'], unitary = True)
+control_thermal.name('control_thermal')
+nodename = control_thermal.name()
+try:
+    control_thermal.hdf5_write(f"{filename_out}",directory = target_directory)
+except:
+    print(f"I had problems writing to the correct file {filename}.h5, so I'm going to try to save your file to temp_ctrl.h5 in the current directory"
+        )
```

Comment:
this is the control for comparison after ensuring there were no microwaves leaking